ons_alpha/core/blocks/embeddable.py

```diff
@@ -14,15 +16,47 @@ class Meta:
         template = "templates/components/streamfield/image_block.html"
 
 
+class DocumentBlockStructValue(blocks.StructValue):
+    def as_macro_data(self):
+        return {
+            "thumbnail": True,
+            "url": self["document"].url,
+            "title": self["title"] or self["document"].filename,
+            "description": self["description"],
+            "metadata": {
+                "file": {
+                    "fileType": self["document"].file_extension.upper(),
+                    "fileSize": filesizeformat(self["document"].get_file_size()),
+                }
+            },
+        }
+
+
 class DocumentBlock(blocks.StructBlock):
     document = DocumentChooserBlock()
     title = blocks.CharBlock(required=False)
+    description = blocks.RichTextBlock(features=settings.RICH_TEXT_BASIC, required=False)
 
     class Meta:
         icon = "doc-full-inverse"
+        value_class = DocumentBlockStructValue
         template = "templates/components/streamfield/document_block.html"
 
 
+class DocumentsBlock(blocks.StreamBlock):
```

Comment:
A `ListBlock` would also have worked here, but subclassing `StreamBlock` is a better-defined path.